app/src/test/java/com/kickstarter/viewmodels/CreatorDashboardReferrerBreakdownHolderViewModelTest.java

```diff
@@ -0,0 +1,369 @@
+  private final TestSubscriber<String> externalReferrerPercentText = new TestSubscriber<>();
+  private final TestSubscriber<Integer> internalReferrerColor = new TestSubscriber<>();
+  private final TestSubscriber<String> internalReferrerPercentText = new TestSubscriber<>();
+  private final TestSubscriber<Boolean> pledgedViaCustomLayoutIsGone = new TestSubscriber<>();
+  private final TestSubscriber<Boolean> pledgedViaExternalLayoutIsGone = new TestSubscriber<>();
+  private final TestSubscriber<Boolean> pledgedViaInternalLayoutIsGone = new TestSubscriber<>();
+  private final TestSubscriber<Pair<Project, Integer>> projectAndAveragePledge = new TestSubscriber<>();
+  private final TestSubscriber<Pair<Project, Float>> projectAndCustomReferrerPledgedAmount = new TestSubscriber<>();
+  private final TestSubscriber<Pair<Project, Float>> projectAndExternalReferrerPledgedAmount = new TestSubscriber<>();
+  private final TestSubscriber<Pair<Project, Float>> projectAndInternalReferrerPledgedAmount = new TestSubscriber<>();
+
+  protected void setUpEnvironment(final @NonNull Environment environment) {
+    this.vm = new CreatorDashboardReferrerBreakdownHolderViewModel.ViewModel(environment);
+    this.vm.outputs.customReferrerPercent().subscribe(this.customReferrerPercent);
+    this.vm.outputs.externalReferrerPercent().subscribe(this.externalReferrerPercent);
+    this.vm.outputs.internalReferrerPercent().subscribe(this.internalReferrerPercent);
+    this.vm.outputs.customReferrerColor().subscribe(this.customReferrerColor);
+    this.vm.outputs.customReferrerPercentText().subscribe(this.customReferrerPercentText);
+    this.vm.outputs.externalReferrerColor().subscribe(this.externalReferrerColor);
+    this.vm.outputs.externalReferrerPercentText().subscribe(this.externalReferrerPercentText);
+    this.vm.outputs.internalReferrerColor().subscribe(this.internalReferrerColor);
+    this.vm.outputs.internalReferrerPercentText().subscribe(this.internalReferrerPercentText);
+    this.vm.outputs.pledgedViaCustomLayoutIsGone().subscribe(this.pledgedViaCustomLayoutIsGone);
+    this.vm.outputs.pledgedViaExternalLayoutIsGone().subscribe(this.pledgedViaExternalLayoutIsGone);
+    this.vm.outputs.pledgedViaInternalLayoutIsGone().subscribe(this.pledgedViaInternalLayoutIsGone);
+    this.vm.outputs.projectAndAveragePledge().subscribe(this.projectAndAveragePledge);
+    this.vm.outputs.projectAndCustomReferrerPledgedAmount().subscribe(this.projectAndCustomReferrerPledgedAmount);
+    this.vm.outputs.projectAndExternalReferrerPledgedAmount().subscribe(this.projectAndExternalReferrerPledgedAmount);
+    this.vm.outputs.projectAndInternalReferrerPledgedAmount().subscribe(this.projectAndInternalReferrerPledgedAmount);
+  }
+
+  @Test
+  public void testCustomReferrerColor() {
+    final Project project = ProjectFactory.project();
+    final ProjectStatsEnvelope statsEnvelope = ProjectStatsEnvelopeFactory.projectStatsEnvelope();
```

Comment:
it doesn't technically matter because the customReferrerColor will always be the same, but we could either add fake custom stats to this envelope to make it clearer that we're testing on custom referrers, OR we could even refactor the viewmodel to only emit these colors when we need them (i.e. takeWhen(customReferrerStats) rather than takeWhen(projectStatsEnvelope))